### PR TITLE
Fixing issue to make it work with geom2d !

### PR DESCRIPTION
**Pull request by UZerbinati**
I was wrong the problem was easier to solve then expected.
Edgenr start counting from 1 if you use Spline geometry (old netgen way)
Edgenr start counting from 0 if you use OCC geometry (new netgen way)

### CODE REVIEW

File: ngsPETSc/plex.py [hunk 1]
```diff
@@ -8,6 +8,10 @@
 from mpi4py import MPI
 import netgen.meshing as ngm
 from netgen.occ import OCCGeometry
+try:
+    from netgen.geom2d import SplineGeometry
```
Comment (pbrubeck) [inline ngsPETSc/plex.py:12]: Why do we need to guard the import?

File: ngsPETSc/plex.py [hunk 2]
```diff
@@ -125,8 +131,10 @@ def __init__(self, mesh=None, comm=None, geo=None, name="Default"):
         if isinstance(mesh, (ngs.comp.Mesh, ngm.Mesh)):
             self.ngMesh, self.petscPlex = self.createPETScDMPlex(mesh)
         elif isinstance(mesh, PETSc.DMPlex):
-            if (geo is not None) and not isinstance(geo, OCCGeometry):
-                raise ValueError("Conversion from DMPlex to Netgen mesh requires OCCGeometry")
+            supportedGeo = (OCCGeometry,) + ((SplineGeometry,) if SplineGeometry else ())
+            if (geo is not None) and not isinstance(geo, supportedGeo):
+                raise ValueError(
```
Comment (pbrubeck) [inline ngsPETSc/plex.py]:
```suggestion
                raise TypeError(
```

File: ngsPETSc/plex.py [hunk 3]
```diff
@@ -125,8 +131,10 @@ def __init__(self, mesh=None, comm=None, geo=None, name="Default"):
         if isinstance(mesh, (ngs.comp.Mesh, ngm.Mesh)):
             self.ngMesh, self.petscPlex = self.createPETScDMPlex(mesh)
         elif isinstance(mesh, PETSc.DMPlex):
-            if (geo is not None) and not isinstance(geo, OCCGeometry):
-                raise ValueError("Conversion from DMPlex to Netgen mesh requires OCCGeometry")
+            supportedGeo = (OCCGeometry,) + ((SplineGeometry,) if SplineGeometry else ())
```
Comment (pbrubeck) [inline ngsPETSc/plex.py]: `supportedGeo` should be defined when we are importing things

**Review by pbrubeck — COMMENTED** [review]
We should add tests ensuring that `SplineGeometry` gets correctly snapped to the boundary. Maybe just parametrize the MG tests to include both OCC and SplineGeometry

File: ngsPETSc/plex.py (hunk 1 above)
Comment (UZerbinati) [reply ngsPETSc/plex.py:12]: Maybe in the future one will able to install Netgen with out Spline geometry but only OCC, since once can do the other way around already.

File: ngsPETSc/plex.py [hunk 4]
```diff
@@ -94,7 +167,9 @@ def createNetgenMesh(plex, geo):
 
         if tdim == 2:
             for face in faces:
-                edge = ngm.Element1D(list(face+1), index=bcLabel, edgenr=bcLabel-1)
+                isSpline = SplineGeometry and isinstance(geo, SplineGeometry)
+                edgenr = bcLabel if isSpline else bcLabel-1
```
Comment (pbrubeck) [inline ngsPETSc/plex.py:171]:
```suggestion
                edgenr = bcLabel-1 if isinstance(goe, OCCGeometry) else bcLabel
```

File: ngsPETSc/plex.py [hunk 5]
```diff
@@ -22,6 +26,52 @@ class comp:
 CELL_SETS_LABEL = "Cell Sets"
 EDGE_SETS_LABEL = "Edge Sets"
 
+supported_geometries = (OCCGeometry,) + ((SplineGeometry,) if SplineGeometry else ())
+
+
```
Comment (pbrubeck) [inline ngsPETSc/plex.py:31]:
```suggestion
```

File: ngsPETSc/plex.py (hunk 1 above)
Comment (pbrubeck) [inline ngsPETSc/plex.py:12]:
```suggestion
    from netgen.geom2d import SplineGeometry
    supported_geometries += (SplineGeometry,)
```

File: ngsPETSc/plex.py [hunk 6]
```diff
@@ -8,6 +8,10 @@
 from mpi4py import MPI
 import netgen.meshing as ngm
 from netgen.occ import OCCGeometry
+try:
```
Comment (pbrubeck) [inline ngsPETSc/plex.py:11]:
```suggestion
from netgen.occ import OCCGeometry
supported_geometries = (OCCGeometry,)
try:
```

**Comment by pbrubeck** [comment]
This only fixes circular arcs, what about general splines?
